app/src/main/java/gov/hhs/cdc/trustedintermediary/auth/AuthDomainRegistration.java

```diff
@@ -0,0 +1,78 @@
+package gov.hhs.cdc.trustedintermediary.auth;
+
+import gov.hhs.cdc.trustedintermediary.context.ApplicationContext;
+import gov.hhs.cdc.trustedintermediary.domainconnector.DomainConnector;
+import gov.hhs.cdc.trustedintermediary.domainconnector.DomainRequest;
+import gov.hhs.cdc.trustedintermediary.domainconnector.DomainResponse;
+import gov.hhs.cdc.trustedintermediary.domainconnector.HttpEndpoint;
+import gov.hhs.cdc.trustedintermediary.wrappers.InvalidTokenException;
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.function.Function;
+import javax.inject.Inject;
+
+/**
+ * The domain connector for the Auth domain. It connects it with the larger trusted intermediary.
+ */
+public class AuthDomainRegistration implements DomainConnector {
+
+    @Inject AuthController authController;
+    @Inject RequestSessionTokenUsecase requestSessionTokenUsecase;
+    @Inject Logger logger;
+
+    private final Map<HttpEndpoint, Function<DomainRequest, DomainResponse>> endpoints =
+            Map.of(new HttpEndpoint("POST", "/v1/auth"), this::handleAuth);
+
+    @Override
+    public Map<HttpEndpoint, Function<DomainRequest, DomainResponse>> domainRegistration() {
+        ApplicationContext.register(AuthController.class, AuthController.getInstance());
+        ApplicationContext.register(
+                RequestSessionTokenUsecase.class, RequestSessionTokenUsecase.getInstance());
+        return endpoints;
+    }
+
+    @Override
+    public String openApiSpecification() {
+        try (InputStream openApiStream =
+                getClass().getClassLoader().getResourceAsStream("openapi_auth.yaml")) {
+            return new String(openApiStream.readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    DomainResponse handleAuth(DomainRequest request) {
+        AuthRequest authRequest;
+        try {
+            authRequest = authController.parseAuthRequest(request);
+        } catch (Exception e) {
+            return authController.constructResponse(400);
+        }
+
+        var token = "";
+        var payload = "";
+
+        try {
+            token = requestSessionTokenUsecase.getToken(authRequest);
+        } catch (InvalidTokenException | UnknownOrganizationException e) {
+            logger.logInfo("Authentication failed", e);
+            return authController.constructResponse(401);
+
+        } catch (Exception e) {
+            logger.logFatal("Bad authentication service configuration: Authentication failed", e);
```

Comment:
I think logError is more appropriate here since the control flow is not interrupted

Reply:
You think so?  This is indicative that we, the engineering team, incorrectly configured or coded something in our application.  When do you think a fatal level should be used?

Reply:
I think it should be used in exceptional cases when the application becomes unstable due to a critical issue that doesn't allow it to continue working properly. For context, the logging library we're using `slf4j` doesn't even include a FATAL level so I had to add it. I thought it could be useful to add in cases where the app needs urgent attention (e.g. in BEDAP, the person on-call would get paged if a fatal error was logged). Here's some more info: https://betterstack.com/community/guides/logging/log-levels-explained/#fatal

Reply:
On second thought, you're right. This would be a critical error that needs our immediate attention. Sorry,  I didn't pay close enough attention

Reply:
Sounds good!